### PR TITLE
I219 incorrect path handling

### PR DESCRIPTION
**Pull request by SerCoach1**
#219

### CODE REVIEW

**Comment by jogobom** [comment]
Not sure why the build failed, it's weird. When I run the tests they're just aborting sometimes for no obvious reason.

**Comment by SerCoach1** [comment]
@jogobom i think it's because the tests execute the "Environment.Exit" part of code and just stop, need to fix that

**Comment by jogobom** [comment]
Yes, it is. I was just about to say that. I hadn't spotted that the tests are now causing Environment.Exit to be called.

**Comment by jogobom** [comment]
Having looked at the code more, I don't think PathManager or RunParameterChecker should be calling Environment.Exit. I apologise for previous statements where I might have said that would be okay. I was wrong. As a user of those classes (and the tests are one of those users) I wouldn't expect them to call Environment.Exit.

**Comment by jogobom** [comment]
I think I said that "exiting directly is fine if you're close to the top of the program (`Main` or maybe one function down from there)". `PathManager.GetFilesInDirectory` is not in that category, it's called from inside check classes. In the case of `RunParameterChecker.ParametersAreValid`, it is called from `Main` but it sounds like a simple check, it doesn't sound like a function that would Exit without warning.

File: YoCode/Program.cs [hunk 1]
```diff
@@ -33,21 +33,10 @@ private static async Task Main(string[] args)
...
-                else
-                {
-                    compositeOutput.ShowHelp();
-                    LaunchReport(result, outputPath);
-                }
-                return;
-            }
+            var parameters = new RunParameterChecker(compositeOutput, appSettingsBuilder);
+
+            parameters.ParametersAreValid(outputPath, result);
```
Comment (jogobom) [inline YoCode/Program.cs]: This function looks weird if you're not checking the return value. It should do what it did before - check the return value and then take appropriate action. `ParametersAreValid` doesn't communicate the severity of what's going to happen.

File: YoCode/RunParameterChecker.cs [hunk 2]
```diff
@@ -20,25 +19,63 @@ internal class RunParameterChecker : IRunParameterChecker
...
+        private void CheckInput(string outputPath, IInputResult result)
         {
             if (result.HelpAsked)
             {
-                return false;
+                compositeOutput.ShowHelp();
+                if (result.CreateHtmlReport && result.OpenHtmlReport)
+                {
+                    WebWriter.LaunchReport(result, outputPath);
+                }
```
Comment (jogobom) [inline YoCode/RunParameterChecker.cs]: I don't think all this ShowHelp and LaunchReport logic should be in this class. It should check the parameters but it shouldn't be the thing that controls what actions are taken.

**Pull request merged**